x/bitcoin/keeper/msg_server.go

```diff
@@ -1092,7 +1096,7 @@ func validateKeyAssignment(ctx sdk.Context, k types.BTCKeeper, signer types.Sign
 	return nil
 }
 
-func getDepositAddress(ctx sdk.Context, k types.BTCKeeper, s types.Signer, key tss.Key, recipient nexus.CrossChainAddress) (types.AddressInfo, error) {
+func getDepositAddress(ctx sdk.Context, k types.BTCKeeper, s types.Signer, key tss.Key, recipient nexus.CrossChainAddress, nonce [sha256.Size]byte) (types.AddressInfo, error) {
```

Comment:
Again, why not calculate nonce here within the function?

Reply:
moved nonce to inside this function